packages/shared-integration/src/context.ts

```diff
@@ -108,7 +108,8 @@ export function createContext<Config extends UserConfig<any> = UserConfig<any>>(
   async function flushTasks() {
     const _tasks = [...tasks]
     await Promise.all(_tasks)
-    tasks.splice(0, _tasks.length)
+    if (tasks[0] === _tasks[0])
```

Comment:
This assumes all the tasks are different, which I think is true after searching in the codebase. It is also possible to make `tasks` a `Set`, but this may reduce performance.